src/yunohost/app.py

```diff
@@ -1497,9 +1509,22 @@ def _parse_args_from_manifest(manifest, action, args={}, auth=None):
                         raise MoulinetteError(errno.EINVAL,
                             m18n.n('app_argument_choice_invalid',
                                 name=arg_name, choices='0, 1'))
-            args_list.append(arg_value)
+            args_list[arg_name] = arg_value
     return args_list
 
+def _make_environment_dict(args_dict):
+    """
+    Convert a dictionnary containing manifest arguments
+    to a dictionnary of env. var. to be passed to scripts
+
+    Keyword arguments:
+        arg -- A key/value dictionnary of manifest arguments
+
+    """
+    env_dict = {}
+    for arg_name, arg_value in args_dict.items():
+        env_dict[ "YNH_APP_ARG_%s" % arg_name ] = arg_value
```

Comment:
I would prefer to format environment variables to be uppercase only in order to be more rigorous, as discussed on the bug tracker (I don't succeed to find the ticket anymore...).